### PR TITLE
Upgrade dependencies

### PR DESCRIPTION
**Pull request by torredil**
#### What type of PR is this?

Maintenance.

#### What is this PR about? / Why do we need it?

A final dependency upgrade ahead of our November release.

The following linter error is also addressed in this PR:
```
pkg/mounter/mount.go:52:6: the methods of "NodeMounter" use pointer receiver and non-pointer receiver. (recvcheck)
type NodeMounter struct {
     ^
```

#### How was this change tested?

```
make update && make verify && make test
```

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: hack/tools/install.sh [hunk 1]
```diff
@@ -19,17 +19,17 @@ set -euo pipefail
...
+GINKGO_VERSION="v2.21.0"
 # https://github.com/golangci/golangci-lint
-GOLANGCI_LINT_VERSION="v1.61.0"
+GOLANGCI_LINT_VERSION="v1.62.0"
 # https://github.com/hairyhenderson/gomplate
-GOMPLATE_VERSION="v4.1.0"
+GOMPLATE_VERSION="v4.2.0"
 # https://github.com/helm/helm
 HELM_VERSION="v3.16.2"
 # https://github.com/kubernetes/kops
```
Comment (ElijahQuinones) [inline hack/tools/install.sh:35]: can we bump the kops version to the latest commit 

80806a8b438fe59dbcbad9a255b890a4ac2c8dd3

https://github.com/kubernetes/kops/commit/80806a8b438fe59dbcbad9a255b890a4ac2c8dd3

Comment (torredil) [reply hack/tools/install.sh:35]: Bumped ✅

**Review by ElijahQuinones — APPROVED** [review]
/lgtm

**Review by AndrewSirenko — APPROVED** [review]

**Pull request merged**